spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/ReactiveResilience4JCircuitBreaker.java

```diff
@@ -98,22 +105,29 @@ public ReactiveResilience4JCircuitBreaker(String id,
 
 	@Override
 	public <T> Flux<T> run(Flux<T> toRun, Function<Throwable, Flux<T>> fallback) {
-		io.github.resilience4j.circuitbreaker.CircuitBreaker defaultCircuitBreaker = circuitBreakerRegistry
-				.circuitBreaker(id, circuitBreakerConfig);
-		circuitBreakerCustomizer.ifPresent(customizer -> customizer.customize(defaultCircuitBreaker));
-		TimeLimiter timeLimiter = timeLimiterRegistry.timeLimiter(id, timeLimiterConfig);
-		Flux<T> toReturn = toRun.transform(CircuitBreakerOperator.of(defaultCircuitBreaker))
-				.timeout(timeLimiter.getTimeLimiterConfig().getTimeoutDuration())
+		Tuple2<CircuitBreaker, TimeLimiter> tuple = buildCircuitBreakerAndTimeLimiter();
+		Flux<T> toReturn = toRun.transform(CircuitBreakerOperator.of(tuple._1()))
+				.timeout(tuple._2().getTimeLimiterConfig().getTimeoutDuration())
 				// Since we are using the Flux timeout we need to tell the circuit breaker
 				// about the error
 				.doOnError(TimeoutException.class,
-						t -> defaultCircuitBreaker.onError(
-								timeLimiter.getTimeLimiterConfig().getTimeoutDuration().toMillis(),
+						t -> tuple._1().onError(tuple._2().getTimeLimiterConfig().getTimeoutDuration().toMillis(),
 								TimeUnit.MILLISECONDS, t));
 		if (fallback != null) {
 			toReturn = toReturn.onErrorResume(fallback);
 		}
 		return toReturn;
 	}
 
+	private Tuple2<CircuitBreaker, TimeLimiter> buildCircuitBreakerAndTimeLimiter() {
```

Comment:
Please remove vavr tuple2. You could replace it with a reactor tuple2 since that is not transitive.